src/shared/yaml-string-builder.spec.ts

```diff
@@ -29,6 +29,15 @@ describe("yaml file builder", () => {
 
     expect(yaml.some.obj.property).toBe("test_value");
   });
+  it("should should update value at path with escaped colon", () => {
```

Comment:
Would it be worth having a test for `a:\\b:c\\d` to check it can handle multiple matches?